src/components/EditableField/EditableField.utils.ts

```diff
@@ -0,0 +1,78 @@
+import { FieldAction, FieldValue } from './EditableField.types'
+import { isArray, isObject } from '../../utilities/is'
+
+export const COMPONENT_KEY = 'EditableField'
+
+export const ACTION_ICONS = {
+  delete: 'cross-medium',
+  link: 'link',
+  plus: 'plus-medium',
+  valueOption: 'chevron-down',
+}
+
+export const deleteAction: FieldAction = {
+  name: 'delete',
+}
+
+export function normalizeFieldValue({
+  value,
+  name,
+  createNewFieldValue,
+  defaultOption,
+}): FieldValue[] {
+  let fieldValue: FieldValue[] = []
+
+  if (isArray(value)) {
+    fieldValue = value.map(val =>
+      createNewFieldValue({ value: val, name }, defaultOption)
+    )
+  } else {
+    fieldValue.push(createNewFieldValue({ value, name }, defaultOption))
+  }
+
+  return fieldValue
```

Comment:
We could go an immutable route here which is a bit more succinct and means we don't need to build up the value of fieldValue over time.

E.g.

```
return isArray(value)
  ? value.map(value => createNewFieldValue({ value, name }, defaultOption)
  : [createNewFieldValue({ value, name }, defaultOption)]
```

Thoughts?

Reply:
I like that!

Reply:
Me too, thanks!